### PR TITLE
refs #54629 admin create edit delete, show all subjects

### PR DESCRIPTION
**Pull request by NeverSmileK57CLC**

### CODE REVIEW

File: app/views/admin/subjects/index.js.erb [hunk 1]
```diff
@@ -0,0 +1 @@
+$("#error_messages").html("<%= j render 'shared/error_messages', object: @subject %>");
```
Comment (ggbond2605) [inline app/views/admin/subjects/index.js.erb]: eof

Comment (NeverSmileK57CLC) [inline app/views/admin/subjects/index.js.erb]: đâu :)

File: app/views/admin/subjects/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  <tbody>
+    <% @subjects.each do |subject| %>
+      <tr>
+        <td><%= subject.id %></td>
+        <td><%= subject.name %></td>
+        <td><%= subject.duration.to_formatted_s(:time) %></td>
+        <td><%= subject.question_num %></td>
+        <td><%= subject.created_at %></td>
+        <td><%= link_to t(".edit"), edit_admin_subject_path(subject), class: "btn btn-warning" %></td>
+        <td><%= link_to t(".delete"), admin_subject_path(subject), method: :delete,
```
Comment (hellovietnam93) [inline app/views/admin/subjects/index.html.erb]: divide into 2 lines

File: app/views/admin/subjects/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+
+<button type="button" class="btn btn-primary" data-toggle="modal" data-target="#myModal">
+  <%= t ".new_subject" %>
+</button>
+
+<div class="modal fade" id="myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
```
Comment (hellovietnam93) [inline app/views/admin/subjects/index.html.erb]: create new file to render

File: app/controllers/admin/subjects_controller.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
+class Admin::SubjectsController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @subject = Subject.new
+    respond_to do |format|
+      format.html
+      format.js {render js: "window.location = '#{admin_subjects_path}'"}
```
Comment (hellovietnam93) [inline app/controllers/admin/subjects_controller.rb]: xem lai cach dung nhe

File: config/routes.rb [hunk 5]
```diff
@@ -8,5 +8,6 @@
   namespace :admin do
     root "users#index"
     resources :users
+    resources :subjects
```
Comment (hellovietnam93) [inline config/routes.rb]: use all method. where is show and new ?

**Comment by hellovietnam93** [comment]
OK

**Pull request merged**